test/generic/FunctionField-test.jl

```diff
@@ -98,14 +98,14 @@ end
       S, y = FunctionField(f, "y")
 
       # TODO: test more than just the result type
-      test_rand(S, 1:10, -10:10)
+      @test_broken false # test_rand(S, 1:10, -10:10) # TODO: make this work again after #1412
    end
-   
+
    for f in P2
       S, y = FunctionField(f, "y")
 
       # TODO: test more than just the result type
-      test_rand(S, 1:10)
+      @test_broken false # test_rand(S, 1:10) # TODO: make this work again after #1412
```

Comment:
I have no idea how and why this broke. If you have an idea, please let me know.

Reply:
Argh I wrote a bunch of comments here and my browser died *sigh*. Let's see if I can cobble it back together:

As far as I can tell, there is already a problem here in `master`... but somehow there I can only trigger it when I handle the test file "interactively". That is, if I enter the following, I get an error also on `master` (I am assuming this code is executed from inside the `AbstractAlgebra.jl` directory:
```
using AbstractAlgebra, Test
include("test/rand.jl")
include("test/generic/FunctionField-test.jl")
```
this gives me:
```
...
Test Summary:                  | Pass  Total  Time
Generic.FunctionField.printing |    9      9  0.0s
Generic.FunctionField.rand: Error During Test at /Users/mhorn/Projekte/OSCAR/AbstractAlgebra.jl/test/generic/FunctionField-test.jl:96
  Got exception outside of a @test
  return type AbstractAlgebra.Generic.FunctionFieldElem{Rational{BigInt}} does not match inferred return type Any
  Stacktrace:
    [1] error(s::String)
      @ Base ./error.jl:35
    [2] test_rand(::Union{Nothing, Function}, ::Any, ::Any, ::Vararg{Any}; type::Any)
      @ Main ~/Projekte/OSCAR/AbstractAlgebra.jl/test/rand.jl:32
    [3] test_rand
      @ ~/Projekte/OSCAR/AbstractAlgebra.jl/test/rand.jl:18 [inlined]
    [4] #test_rand#9
      @ ~/Projekte/OSCAR/AbstractAlgebra.jl/test/rand.jl:15 [inlined]
    [5] test_rand(::Any, ::Any, ::Any)
      @ Main ~/Projekte/OSCAR/AbstractAlgebra.jl/test/rand.jl:15
    [6] macro expansion
      @ ~/Projekte/OSCAR/AbstractAlgebra.jl/test/generic/FunctionField-test.jl:101 [inlined]
    [7] macro expansion
      @ ~/.julia/juliaup/julia-1.9.3+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/Test/src/Test.jl:1498 [inlined]
    [8] top-level scope
      @ ~/Projekte/OSCAR/AbstractAlgebra.jl/test/generic/FunctionField-test.jl:97
    [9] include(fname::String)
      @ Base.MainInclude ./client.jl:478
   [10] top-level scope
      @ REPL[17]:1
   [11] eval
      @ ./boot.jl:370 [inlined]
   [12] eval_user_input(ast::Any, backend::REPL.REPLBackend, mod::Module)
      @ REPL ~/.julia/juliaup/julia-1.9.3+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:153
   [13] repl_backend_loop(backend::REPL.REPLBackend, get_module::Function)
      @ REPL ~/.julia/juliaup/julia-1.9.3+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:249
   [14] start_repl_backend(backend::REPL.REPLBackend, consumer::Any; get_module::Function)
      @ REPL ~/.julia/juliaup/julia-1.9.3+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:234
   [15] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool, backend::Any)
      @ REPL ~/.julia/juliaup/julia-1.9.3+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:379
   [16] run_repl(repl::REPL.AbstractREPL, consumer::Any)
      @ REPL ~/.julia/juliaup/julia-1.9.3+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/REPL/src/REPL.jl:365
   [17] (::Base.var"#1017#1019"{Bool, Bool, Bool})(REPL::Module)
      @ Base ./client.jl:421
   [18] #invokelatest#2
      @ ./essentials.jl:819 [inlined]
   [19] invokelatest
      @ ./essentials.jl:816 [inlined]
   [20] run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)
      @ Base ./client.jl:405
   [21] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:322
   [22] _start()
      @ Base ./client.jl:522
Test Summary:              | Error  Total  Time
Generic.FunctionField.rand |     1      1  0.1s
ERROR: LoadError: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /Users/mhorn/Projekte/OSCAR/AbstractAlgebra.jl/test/generic/FunctionField-test.jl:96
```


And this I tracked down further to
```
using AbstractAlgebra, Test
using AbstractAlgebra.RandomExtensions: RandomExtensions, make
R = polynomial_ring(polynomial_ring(QQ,:x1)[1], :y)[1]
@inferred make(R, 0:0, 1:10, -10:10)
```
But `@code_warntype make(R, 0:0, 1:10, -10:10)` reveals nothing, no code ambiguity?  It also infers the correct result type. Huh, weird.

Reply:
PR #1420 fixes this, so in theory after merging this we can re-run the tests here and hopefully they'll pass

Reply:
Now that #1420 is merged, I reverted the commenting out. Let's hope it succeeds now.